From: Yifeng <wuyifeng@nd.com.cn>
Date: Tue, 2 Aug 2022 07:40:00 +0000
Subject: Enable Javascript JIT, WebGL and WebRTC by default

Fix version: 1.2.1
Enable Javascript JIT, WebGL, WebRTC and SharedArrayBuffer by default.
This patch is to fix the following issues:
- AP9-9472
- AP9-9914
- AP9-8071

---
 components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteCustomContentSetting.java | 10 +++++-----
 components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteCustomContentSettingImpl.java | 2 +-
 components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteWebGLContentSetting.java | 4 ++--
 components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteWebRTCContentSetting.java | 4 ++--
 components/content_settings/core/browser/content_settings_registry.cc | 6 +++---
 content/public/common/content_features.cc | 6 +++---
 6 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteCustomContentSetting.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteCustomContentSetting.java
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteCustomContentSetting.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteCustomContentSetting.java
@@ -33,7 +33,7 @@ public abstract class BromiteCustomContentSetting {
 
     private @ContentSettingsType int mContentSettingsType;
     private @SiteSettingsCategory.Type int mSiteSettingsCategory;
-    private @ContentSettingValues Integer mInitialDefaultValue;
+    private @ContentSettingValues Integer mDefaultEnabledValue;
     private @ContentSettingValues Integer mDefaultDisabledValue;
     private boolean mAllowException;
     private String mPreferenceKey;
@@ -41,14 +41,14 @@ public abstract class BromiteCustomContentSetting {
 
     public BromiteCustomContentSetting(@ContentSettingsType int contentSettingsType,
                                        @SiteSettingsCategory.Type int siteSettingsCategory,
-                                       @ContentSettingValues Integer initialDefaultValue,
+                                       @ContentSettingValues Integer defaultEnabledValue,
                                        @ContentSettingValues Integer defaultDisabledValue,
                                        boolean allowException,
                                        String preferenceKey,
                                        String profilePrefKey) {
         mContentSettingsType = contentSettingsType;
         mSiteSettingsCategory = siteSettingsCategory;
-        mInitialDefaultValue = initialDefaultValue;
+        mDefaultEnabledValue = defaultEnabledValue;
         mDefaultDisabledValue = defaultDisabledValue;
         mAllowException = allowException;
         mPreferenceKey = preferenceKey;
@@ -63,8 +63,8 @@ public abstract class BromiteCustomContentSetting {
         return mSiteSettingsCategory;
     }
 
-    protected @ContentSettingValues Integer getInitialDefaultValue() {
-        return mInitialDefaultValue;
+    protected @ContentSettingValues Integer getDefaultEnabledValue() {
+        return mDefaultEnabledValue;
     }
 
     public @ContentSettingValues Integer getDefaultDisabledValue() {
diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteCustomContentSettingImpl.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteCustomContentSettingImpl.java
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteCustomContentSettingImpl.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteCustomContentSettingImpl.java
@@ -127,7 +127,7 @@ public abstract class BromiteCustomContentSettingImpl {
             int default_value = WebsitePreferenceBridge.getDefaultContentSetting(
                                     browserContextHandle, cs.getContentSetting());
             if (SingleCategorySettings.BINARY_TOGGLE_KEY.equals(preference.getKey())) {
-                int setting = ((boolean) newValue) == true ? default_value :
+                int setting = ((boolean) newValue) == true ? cs.getDefaultEnabledValue() :
                                                              cs.getDefaultDisabledValue();
 
                 WebsitePreferenceBridge.setDefaultContentSetting(browserContextHandle,
diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteWebGLContentSetting.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteWebGLContentSetting.java
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteWebGLContentSetting.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteWebGLContentSetting.java
@@ -33,7 +33,7 @@ public class BromiteWebGLContentSetting extends BromiteCustomContentSetting {
     public BromiteWebGLContentSetting() {
         super(/*contentSettingsType*/ ContentSettingsType.WEBGL,
               /*siteSettingsCategory*/ SiteSettingsCategory.Type.WEBGL,
-              /*initialDefaultValue*/ ContentSettingValues.BLOCK,
+              /*defaultEnabledValue*/ ContentSettingValues.ALLOW,
               /*defaultDisabledValue*/ ContentSettingValues.BLOCK,
               /*allowException*/ true,
               /*preferenceKey*/ "webgl",
@@ -45,7 +45,7 @@ public class BromiteWebGLContentSetting extends BromiteCustomContentSetting {
         return new ContentSettingsResources.ResourceItem(
             /*icon*/ R.drawable.web_asset,
             /*title*/ R.string.webgl_permission_title,
-            /*defaultEnabledValue*/ getInitialDefaultValue(),
+            /*defaultEnabledValue*/ getDefaultEnabledValue(),
             /*defaultDisabledValue*/ getDefaultDisabledValue(),
             /*enabledSummary*/ R.string.website_settings_category_webgl_enabled,
             /*disabledSummary*/ R.string.website_settings_category_webgl_disabled);
diff --git a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteWebRTCContentSetting.java b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteWebRTCContentSetting.java
--- a/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteWebRTCContentSetting.java
+++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/BromiteWebRTCContentSetting.java
@@ -33,7 +33,7 @@ public class BromiteWebRTCContentSetting extends BromiteCustomContentSetting {
     public BromiteWebRTCContentSetting() {
         super(/*contentSettingsType*/ ContentSettingsType.WEBRTC,
               /*siteSettingsCategory*/ SiteSettingsCategory.Type.WEBRTC,
-              /*initialDefaultValue*/ ContentSettingValues.BLOCK,
+              /*defaultEnabledValue*/ ContentSettingValues.ALLOW,
               /*defaultDisabledValue*/ ContentSettingValues.BLOCK,
               /*allowException*/ true,
               /*preferenceKey*/ "webrtc",
@@ -45,7 +45,7 @@ public class BromiteWebRTCContentSetting extends BromiteCustomContentSetting {
         return new ContentSettingsResources.ResourceItem(
             /*icon*/ R.drawable.web_asset,
             /*title*/ R.string.webrtc_permission_title,
-            /*defaultEnabledValue*/ getInitialDefaultValue(),
+            /*defaultEnabledValue*/ getDefaultEnabledValue(),
             /*defaultDisabledValue*/ getDefaultDisabledValue(),
             /*enabledSummary*/ R.string.website_settings_category_webrtc_enabled,
             /*disabledSummary*/ R.string.website_settings_category_webrtc_disabled);
diff --git a/components/content_settings/core/browser/content_settings_registry.cc b/components/content_settings/core/browser/content_settings_registry.cc
--- a/components/content_settings/core/browser/content_settings_registry.cc
+++ b/components/content_settings/core/browser/content_settings_registry.cc
@@ -606,7 +606,7 @@ void ContentSettingsRegistry::Init() {
            ContentSettingsInfo::EXCEPTIONS_ON_SECURE_ORIGINS_ONLY);
 
   Register(ContentSettingsType::JAVASCRIPT_JIT, "javascript-jit",
-           CONTENT_SETTING_BLOCK, WebsiteSettingsInfo::UNSYNCABLE,
+           CONTENT_SETTING_ALLOW, WebsiteSettingsInfo::UNSYNCABLE,
            AllowlistedSchemes(),
            ValidSettings(CONTENT_SETTING_ALLOW, CONTENT_SETTING_BLOCK),
            WebsiteSettingsInfo::SINGLE_ORIGIN_ONLY_SCOPE,
@@ -668,7 +668,7 @@ void ContentSettingsRegistry::Init() {
            ContentSettingsInfo::PERSISTENT,
            ContentSettingsInfo::EXCEPTIONS_ON_SECURE_ORIGINS_ONLY);
 
-  Register(ContentSettingsType::WEBGL, "webgl", CONTENT_SETTING_BLOCK,
+  Register(ContentSettingsType::WEBGL, "webgl", CONTENT_SETTING_ALLOW,
            WebsiteSettingsInfo::SYNCABLE,
            AllowlistedSchemes(),
            ValidSettings(CONTENT_SETTING_ALLOW,
@@ -682,7 +682,7 @@ void ContentSettingsRegistry::Init() {
            /*permission_type_ui*/ IDS_SITE_SETTINGS_TYPE_WEBGL,
            /*permission_type_ui_mid_sentence*/ IDS_SITE_SETTINGS_TYPE_WEBGL_MID_SENTENCE);
 
-  Register(ContentSettingsType::WEBRTC, "webrtc", CONTENT_SETTING_BLOCK,
+  Register(ContentSettingsType::WEBRTC, "webrtc", CONTENT_SETTING_ALLOW,
            WebsiteSettingsInfo::SYNCABLE,
            AllowlistedSchemes(),
            ValidSettings(CONTENT_SETTING_ALLOW,
diff --git a/content/public/common/content_features.cc b/content/public/common/content_features.cc
--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -832,12 +832,12 @@ const base::Feature kServiceWorkerTerminationOnNoControllee{
 // This feature is also enabled independently of this flag for cross-origin
 // isolated renderers.
 const base::Feature kSharedArrayBuffer{"SharedArrayBuffer",
-                                       base::FEATURE_DISABLED_BY_DEFAULT};
+                                       base::FEATURE_ENABLED_BY_DEFAULT};
 // If enabled, SharedArrayBuffer is present and can be transferred on desktop
 // platforms. This flag is used only as a "kill switch" as we migrate towards
 // requiring 'crossOriginIsolated'.
 const base::Feature kSharedArrayBufferOnDesktop{
-    "SharedArrayBufferOnDesktop", base::FEATURE_DISABLED_BY_DEFAULT};
+    "SharedArrayBufferOnDesktop", base::FEATURE_ENABLED_BY_DEFAULT};

 // Signed Exchange Reporting for distributors
 // https://www.chromestatus.com/feature/5687904902840320
@@ -1021,7 +1021,7 @@ const base::Feature kTrustedDOMTypes{"TrustedDOMTypes",
 // https://developer.chrome.com/origintrials/#/view_trial/303992974847508481
 // https://crbug.com/1144104
 const base::Feature kUnrestrictedSharedArrayBuffer{
-    "UnrestrictedSharedArrayBuffer", base::FEATURE_DISABLED_BY_DEFAULT};
+    "UnrestrictedSharedArrayBuffer", base::FEATURE_ENABLED_BY_DEFAULT};

 // Allows user activation propagation to all frames having the same origin as
 // the activation notifier frame.  This is an intermediate measure before we
--